components/encryption/src/master_key/kms.rs

```diff
@@ -256,19 +323,32 @@ impl KmsBackend {
             other => {
                 return Err(box_err!(
                     "KMS vendor mismatch expect {:?} got {:?}",
-                    AWS_KMS_VENDOR_NAME,
+                    vendor_name,
                     other
                 ))
             }
         }
 
-        let mut inner = self.inner.lock().unwrap();
-        let ciphertext_key = content.metadata.get(MetadataKey::KmsCiphertextKey.as_str());
-        if ciphertext_key.is_none() {
-            return Err(box_err!("KMS ciphertext key not found"));
+        let ciphertext_key = match content.metadata.get(MetadataKey::KmsCiphertextKey.as_str()) {
+            None => return Err(box_err!("KMS ciphertext key not found")),
+            Some(key) => EncryptedKey::new(key.to_vec())?,
+        };
+
+        let mut state = self.state.lock().unwrap();
```

Comment:
scoping the mutex lock for readablity:
```
{
    let mut state = self.state.lock().unwrap();
    // the rest of code that uses the state
}
```

Reply:
it's the entire rest of the function so the behavior and the compiler analysis is exactly the same. Is this better for developers to read the code? It is also possible it will be confusing to someone reading the code since they don't have any effect.

Reply:
It is a common practices to scope the lock guard for readability. It make it crystal clear how far is the critical session.

Reply:
for example in an later iteration people add more unrelated logic at the end of this function, and may forget to unlock the mutex. explicit scoping avoids that.